### PR TITLE
sqld: Merge http endpoints

### PR DESCRIPTION
**Pull request by LucioFranco**
This commit merges all http endpoints to be under the same socketaddr. This allows us to only expose one port for our external APIs. In addition, this commit adds auth for the replication endpoints.

### CODE REVIEW

File: sqld/src/http/either.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+use http_body::{Body, SizeHint};
+use proj::EitherProj;
+
+/// Sum type with two cases: [`Left`] and [`Right`], used if a body can be one of
+/// two distinct types.
+///
+/// [`Left`]: Either::Left
+/// [`Right`]: Either::Right
+#[derive(Debug, Clone, Copy)]
+pub enum Either<L, R> {
```
Comment (MarinPostma) [inline sqld/src/http/either.rs:20]: we already have transitive dependency to either, might as well bring in the crate

Comment (psarna) [reply sqld/src/http/either.rs:20]: @MarinPostma pretty sure it was needed to be able to `impl Body for Either` below? But I really hope we can do without it if we migrate everything to axum :crossed_fingers:

Comment (MarinPostma) [reply sqld/src/http/either.rs:20]: ah yeah, orphan rule...

Comment (LucioFranco) [reply sqld/src/http/either.rs:20]: This is a special one since it needs the body impl that I don't think the other one has?

**Comment by LucioFranco** [comment]
I am going to close this as, I am going to convert our current API to Axum and move the replication endpoints (read/write) to gRPC which makes a lot of this not needed.

**Pull request closed without merging**